api/v1alpha1/storageclient_types.go

```diff
@@ -52,6 +52,9 @@ type StorageClientStatus struct {
 
 	// ConsumerID will hold the identity of this cluster inside the attached provider cluster
 	ConsumerID string `json:"id,omitempty"`
+
+	// Name of the subscription channel that this client wants the operator to be at.
+	DesiredOperatorSubscriptionChannel string `json:"subscriptionChannel,omitempty"`
```

Comment:
We should stay away from field names that do not match the JSON/YAML prop name. 
Let's settle down on DesiredSubscriptionChannel. In addition, is this something that could be reconstructed if the status is wiped? if so, from where? if notI think we need a different approach.

Reply:
> if notI think we need a different approach.
- ok, may I know why do we need to solve that?

Reply:
moved to storing the channel in annotation.